diff --git a/util-linux.SlackBuild b/util-linux.SlackBuild
index edce684..87448be 100644
--- a/util-linux.SlackBuild
+++ b/util-linux.SlackBuild
@@ -58,6 +58,9 @@ if [ "$ARCH" = "i686" ]; then
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -march=x86-64 -mtune=generic -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "ia64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX=""
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
