diff --git a/fig2dev.SlackBuild b/fig2dev.SlackBuild
index f462a0a..f1675aa 100644
--- a/fig2dev.SlackBuild
+++ b/fig2dev.SlackBuild
@@ -57,6 +57,9 @@ if [ "$ARCH" = "i686" ]; then
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -march=x86-64 -mtune=generic -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "ia64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX=""
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
