diff --git a/libjpeg-turbo.SlackBuild b/libjpeg-turbo.SlackBuild
index 693a0c4..f7fa230 100644
--- a/libjpeg-turbo.SlackBuild
+++ b/libjpeg-turbo.SlackBuild
@@ -53,6 +53,9 @@ if [ "$ARCH" = "i686" ]; then
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -march=x86-64 -mtune=generic -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "ia64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX=""
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -92,6 +92,17 @@ cd build
   make install DESTDIR=$PKG || exit 1
 cd ..
 
+# move `/usr/lib64/*` to `/usr/lib` for ia64, as `-DLIB_SUFFIX` is ignored
+if [ "$ARCH" = "ia64" ]; then
+  if [ -d $PKG/usr/lib64 ]; then
+    if [ -d $PKG/usr/lib ]; then
+      mv $PKG/usr/lib64/* $PKG/usr/lib/ && rm -rf $PKG/usr/lib64
+    else
+      mv $PKG/usr/lib64 $PKG/usr/lib
+    fi
+  fi
+fi
+
 # Don't ship .la files:
 rm -f $PKG/{,usr/}lib${LIBDIRSUFFIX}/*.la
 
