diff --git a/git.SlackBuild b/git.SlackBuild
index ef14e05..539ee40 100644
--- a/git.SlackBuild
+++ b/git.SlackBuild
@@ -64,6 +64,8 @@ elif [ "$ARCH" = "i686" ]; then
   SLKCFLAGS="-O2 -march=pentium4 -mtune=generic"
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -march=x86-64 -mtune=generic -fPIC"
+elif [ "$ARCH" = "ia64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
 else
   SLKCFLAGS="-O2"
 fi
@@ -106,6 +108,7 @@ make $NUMJOBS \
   INSTALLDIRS=vendor \
   ASCIIDOC8=YesPlease \
   USE_LIBPCRE2=YesPlease \
+  NO_RUST=YesPlease \
   all doc || exit 1
 
 make \
@@ -115,6 +118,7 @@ make \
   INSTALLDIRS=vendor \
   ASCIIDOC8=YesPlease \
   USE_LIBPCRE2=YesPlease \
+  NO_RUST=YesPlease \
   install \
   install-doc \
   DESTDIR=$PKG || exit 1
@@ -127,6 +131,7 @@ make -C contrib/subtree \
   INSTALLDIRS=vendor \
   ASCIIDOC8=YesPlease \
   USE_LIBPCRE2=YesPlease \
+  NO_RUST=YesPlease \
   install \
   install-doc \
   DESTDIR=$PKG || exit 1
