diff --git a/mariadb.SlackBuild b/mariadb.SlackBuild
index f32ebf2..06ecb52 100644
--- a/mariadb.SlackBuild
+++ b/mariadb.SlackBuild
@@ -75,6 +75,9 @@ if [ "$ARCH" = "i686" ]; then
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -march=x86-64 -mtune=generic -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "ia64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX=""
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -154,6 +157,25 @@ rm -rf $PKG/etc/systemd
 # Leave build directory:
 cd ..
 
+# move `/usr/lib64/*` to `/usr/lib` for ia64, if existing
+if [ "$ARCH" = "ia64" ]; then
+  if [ -d $PKG/usr/lib64 ]; then
+    if [ -d $PKG/usr/lib ]; then
+      mv $PKG/usr/lib64/* $PKG/usr/lib/ && rm -rf $PKG/usr/lib64
+    else
+      mv $PKG/usr/lib64 $PKG/usr/lib
+    fi
+  fi
+# also do that similarly for `/lib64`, if existing
+  if [ -d $PKG/lib64 ]; then
+    if [ -d $PKG/lib ]; then
+      mv $PKG/lib64/* $PKG/lib/ && rm -rf $PKG/lib64
+    else
+      mv $PKG/lib64 $PKG/lib
+    fi
+  fi
+fi
+
 # Install innotop
 install -m755 debian/additions/innotop/innotop $PKG/usr/bin/
 install -m644 debian/additions/innotop/innotop.1 $PKG/usr/man/man1/
