diff --git a/libbytesize.SlackBuild b/libbytesize.SlackBuild
index d051b76..f3c4ce2 100644
--- a/libbytesize.SlackBuild
+++ b/libbytesize.SlackBuild
@@ -54,6 +54,9 @@ if [ "$ARCH" = "i686" ]; then
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -march=x86-64 -mtune=generic -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "ia64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX=""
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
