diff --git a/python-build.SlackBuild b/python-build.SlackBuild
index 3c7b3c7..a4fb8f1 100644
--- a/python-build.SlackBuild
+++ b/python-build.SlackBuild
@@ -50,6 +50,9 @@ if [ "$ARCH" = "i686" ]; then
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -march=x86-64 -mtune=generic -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "ia64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX=""
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
