diff --git a/libusb.SlackBuild b/libusb.SlackBuild
index d291d80..0f33761 100644
--- a/libusb.SlackBuild
+++ b/libusb.SlackBuild
@@ -53,6 +53,9 @@ if [ "$ARCH" = "i686" ]; then
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -march=x86-64 -mtune=generic -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "ia64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX=""
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -134,4 +137,3 @@ cat $CWD/slack-desc > $PKG/install/slack-desc
 
 cd $PKG
 /sbin/makepkg -l y -c n $TMP/libusb-$VERSION-$ARCH-$BUILD.txz
-
