diff --git a/tree.SlackBuild b/tree.SlackBuild
index ab7a913..f4b94c7 100644
--- a/tree.SlackBuild
+++ b/tree.SlackBuild
@@ -56,6 +56,9 @@ if [ "$ARCH" = "i686" ]; then
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -march=x86-64 -mtune=generic -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "ia64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX=""
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -74,7 +77,7 @@ find . \
   \( -perm 666 -o -perm 664 -o -perm 600 -o -perm 444 -o -perm 440 -o -perm 400 \) \
   -exec chmod 644 {} \+
 
-sed -i "s,CFLAGS=-ggdb,CFLAGS=\$(XCFLAGS) -ggdb,g" Makefile
+sed -i "s,CFLAGS?=-O3,CFLAGS=\$(XCFLAGS),g" Makefile
 make XCFLAGS="$SLKCFLAGS" $NUMJOBS || make || exit 1
 
 mkdir -p $PKG/usr/bin
