diff --git a/grub.SlackBuild b/grub.SlackBuild
index accc311..f51b05f 100644
--- a/grub.SlackBuild
+++ b/grub.SlackBuild
@@ -63,6 +63,10 @@ elif [ "$ARCH" = "x86_64" ]; then
   EFI32_FLAGS=" --with-platform=efi --target=i386 --program-prefix= "
   EFI_FLAGS=" --with-platform=efi --target=x86_64 --program-prefix= "
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "ia64" ]; then
+  SLKCFLAGS=""
+  EFI_FLAGS=" --with-platform=efi --target=ia64 --program-prefix= "
+  LIBDIRSUFFIX=""
 else
   SLKCFLAGS=""
   LIBDIRSUFFIX=""
@@ -124,6 +128,13 @@ cat $CWD/0005-GRUB_DISPLAY_KERNEL_FILETYPES.patch | patch -p1 --verbose || exit
 # standard files in here will be overwritten with every upgrade:
 cat $CWD/0006-warn-grub.d-overwrites-docs.patch | patch -p1 --verbose || exit 1
 
+# Add patches to allow a GRUB standalone image to work with ski-efi
+# See https://github.com/trofi/ski/issues/12
+if [ "$ARCH" = "ia64" ]; then
+  cat $CWD/9998-ia64-Reset-grub_errno-on-failure-to-allocate.patch | patch -p1 --verbose || exit 1
+  cat $CWD/9999-ia64-Disable-optimizations-using-floating-point-arithmetics.patch | patch -p1 --verbose || exit 1
+fi
+
 # Still won't compile. Someday they'll remember how to cut a new release.
 touch grub-core/extra_deps.lst
 
