diff --git a/polkit.SlackBuild b/polkit.SlackBuild
index 93c89c0..44caacb 100644
--- a/polkit.SlackBuild
+++ b/polkit.SlackBuild
@@ -57,6 +57,9 @@ if [ "$ARCH" = "i686" ]; then
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -march=x86-64 -mtune=generic -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "ia64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX=""
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
