diff --git a/wget2.SlackBuild b/wget2.SlackBuild
index 58db113..01f8fa2 100644
--- a/wget2.SlackBuild
+++ b/wget2.SlackBuild
@@ -52,6 +52,9 @@ if [ "$ARCH" = "i686" ]; then
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -march=x86-64 -mtune=generic -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "ia64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX=""
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
