diff --git a/sysstat.SlackBuild b/sysstat.SlackBuild
index fc61cb9..9d8d411 100644
--- a/sysstat.SlackBuild
+++ b/sysstat.SlackBuild
@@ -66,6 +66,10 @@ elif [ "$ARCH" = "arm" ]; then
   SLKCFLAGS="-O2 -march=armv4t"
   LIBDIRSUFFIX=""
   ARCHQUADLET="-gnueabi"
+elif [ "$ARCH" = "ia64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX=""
+  ARCHQUADLET=""
 else
   SLKCFLAGS="-O2"
   ARCHQUADLET=""
@@ -168,4 +172,3 @@ cat $CWD/slack-desc > $PKG/install/slack-desc
 
 cd $TMP/package-${PKGNAM}
 makepkg -l y -c n $TMP/${PKGNAM}-$VERSION-$ARCH-$BUILD.txz
-
